src/extensions/surveys.tsx

```diff
@@ -956,8 +975,16 @@ export function Questions({
     const textColor = getContrastingTextColor(
         survey.appearance?.backgroundColor || defaultSurveyAppearance.backgroundColor
     )
-    const [questionsResponses, setQuestionsResponses] = useState({})
-    const { previewPageIndex, onPopupSurveyDismissed, isPopup, onPreviewSubmit } = useContext(SurveyContext)
+    // Initialize responses from localStorage or empty object
+    const [questionsResponses, setQuestionsResponses] = useState(() => {
+        const inProgressSurveyData = getInProgressSurveyState(survey)
+        if (inProgressSurveyData?.responses) {
+            logger.info('Survey is already in progress, filling in initial responses')
+        }
+        return inProgressSurveyData?.responses || {}
+    })
```

Comment:
restore store the state of the question responses based on the data of a survey that's in progress